code/datums/components/radioactive.dm

```diff
@@ -0,0 +1,77 @@
+#define RAD_AMOUNT_LOW 50
+#define RAD_AMOUNT_MEDIUM 200
+#define RAD_AMOUNT_HIGH 500
+#define RAD_AMOUNT_EXTREME 1000
+
+/datum/component/radioactive
+	dupe_mode = COMPONENT_DUPE_UNIQUE
+
+	var/hl3_release_date //the half-life measured in ticks
+	var/strength
+	var/can_contaminate
+
+/datum/component/radioactive/Initialize(_strength=0, _half_life=RAD_HALF_LIFE, _can_contaminate=TRUE)
+	. = ..()
+	strength = _strength
+	hl3_release_date = _half_life
+	can_contaminate = _can_contaminate
+
+	if(istype(parent, /atom)) 
+		RegisterSignal(COMSIG_PARENT_EXAMINE, .proc/rad_examine)
+		if(istype(parent, /obj/item))
+			RegisterSignal(COMSIG_ITEM_ATTACK, .proc/rad_attack)
+			RegisterSignal(COMSIG_ITEM_ATTACK_OBJ, .proc/rad_attack)
+	else
+		warning("Something that wasn't an atom was given /datum/component/radioactive")
+		qdel(src)
```

Comment:
CRASH instead of warning. I still haven't handled components deleting themselves in Initialize(). Not sure if I should